Removed Gdot character from Huggingface tokenizer

JosselinSomervilleRoberts opened the pull request:
Fixed https://github.com/stanford-crfm/helm/issues/1516.
Changed all the tests and made sure they still pass.
I did not change the `.convert_ids_to_tokens` in `HugginFaceCLient.serve_request()` as it did not seem necessary and it was causing the `gpt2` test in `test_gpt2_window_service.py` to fail.

I will have to rebase this if we push the Anthropic and Palmyra PR (or the other way around).

JosselinSomervilleRoberts commented:
I created #1522 which is related.

yifanmai commented on src/helm/benchmark/window_services/test_t0pp_window_service.py:
```diff
@@ -87,62 +87,62 @@ def test_decode(self):
-            "▁on",
-            "▁Foundation",
-            "▁Model",
+            "The",
```
Could we exclude the `_` tokenizers from this PR? You might have to add a branch in huggingface_client to do that i.e. special-case GPT-2...

This strips the leading spaces from tokens, which is possibly worse. This means that if you concat the tokens, you get "TheCenterforResearch...". Also it modifies the byte count and thus the metrics.

I haven't tested this yet, but I expect that: `convert_ids_to_string([37, 1166])` gives "The Center" (with space) and `convert_ids_to_string([1166])` alone gives "Center" (without space).

JosselinSomervilleRoberts replied:
Ok I added a special case. This is not super clean but it should work. Regarding the `_` what should I do? Replace it manually?

yifanmai commented:
I merged the Anthropic / Writer PRs, so you can rebase now.

JosselinSomervilleRoberts commented:
I have rebased and reverted everything related to "▁". I will wait for the tests to finish before merging. I will address the underscore character in a next PR.
